sycl/include/sycl/ext/oneapi/experimental/cuda/builtins.hpp

```diff
@@ -66,146 +81,387 @@ ldg(const T *ptr) {
     return __nvvm_ldg_ul(ptr);
   } else if constexpr (std::is_same_v<T, unsigned long long>) {
     return __nvvm_ldg_ull(ptr);
+  } else if constexpr (std::is_same_v<T, half>) {
+    auto native = reinterpret_cast<const __fp16 *>(ptr);
+    return __nvvm_ldg_h(native);
```

Comment:
Does this compile for default sm_50? It is annoying because sm_50 is probably rare and now deprecated at Hopper, but ptx docs say we need sm_53 for half tex cache. Maybe we need a arch check here.

Reply:
Yeap, I've compiled it for sm_50 without any problems.